### DISCUSSION TITLE
Idea: Domain specification for LinearModel, QuadraticModel,  ParablolicModel and PolynomialModel

### DISCUSSION THREAD

**Discussion by Tillsten** [lmfit/lmfit-py#847] [Ideas]
One common use of the models above is use them to model backgrounds in spectra. Here, the independent variable is often far away from zero, which sometimes makes the polynomial models above numerically unstable. Therefore an option to set the domain (scale and shift) of the models would be helpful. The numpy polynomial classes also have this feature.

**Comment by newville** [comment]
Thanks. 
Do you think it would be sufficient to use
`[min(x), max(x)]` as the domain?   Would that be better as something we just always do (arguing that it is never worse) or should that be a default and a user option?

I cannot think of any other built-in models that we support that would have such an adaptable feature based on the independent data.  I'm not entirely sure of the best way to do that. It might need some sort of "initialize model" method that gets called at the beginning of each fit, or if `x` has changed?  Not sure...

**Reply by Tillsten** [reply]
I think it would be sufficent, since if you really care about some special domain you will transform your independent into that domain anyways, but who knows what other people are doing. I imagine it could be implented similar to the degree argument of the polynomial class, e.g. at the model initialization.  To be honest I am not sure if such a change would be backward comtible or not.

**Reply by newville** [reply]
I might suggest that we allow this to be a goal/milestone for 1.3.0.
I'm not certain that adding an "initialize model with data" is the only solution.  If there aren't obvious problems currently and this is mostly a sort of generic "improving numerical stability", then we should a) do it, and b) allow ourselves some time to figure it out.

Would that be OK @Tillsten ?

**Reply by Tillsten** [reply]
No need to hurry, I can work around the issue. Since I had exactly this issue multiple times recently, I just wanted to see if this kind of solution is of general interest.
